### ISSUE TITLE
[BUG] upgrade to 2.4.0 from 1.3.6 seems to not check for legacy indices before upgrading structure to lucene 9

### ISSUE THREAD

**Issue by maniel** [opensearch-project/OpenSearch#5540]
**Describe the bug**
It seems that data are somehow upgraded to lucene 9 even if opensearch 2.4.0 refuses to run because legacy indices

**To Reproduce**
Tried to upgrade my single node in place from 1.3.6 to 2.4.0 (docker), new version complained about old indices indexed by ES 6.8.11,
so i downgraded, deleted some indices in 1.3.6, apparently not sufficiently because some of them were left, and 2.4.0 complained again, went back to 1.3.6, which failed to run because:
`java.lang.IllegalArgumentException: indexCreatedVersionMajor is in the future: 9`

**Expected behavior**
shouldn't touch the data after finding legacy indices

**Host/Environment (please complete the following information):**
 - docker-ce 20.10.18
 - OS: Ubuntu
 - Version: 20.04.05 LTS 

**Additional context**
[My forum post about the issue](https://forum.opensearch.org/t/failed-upgrade-from-1-3-6-to-2-4-0-but-cannot-go-back/11737)

**Comment by dblock** [comment]
What was your original data before any attempted upgrades? Is this easily reproducible?

**Comment by anasalkouz** [comment]
Hi @maniel, did you try to recover and spin up new 1.3.6 cluster using an old snapshot? because once you upgrade to 2.4 and you did any indexing operation, it will be hard to downgrade to an older version.

@kartg @sumobrian Could you please take a look?

**Comment by maniel** [comment]
thanks for reply @anasalkouz, that would be my go to if i'd had a snapshot, heck, i could make a filesystem-level snapshot because it's on BTRFS, just didn't think about it, i mean every ES upgrade (or from ES to OS) was seamless, if new one wouldn't start because of legacy indices i'd delete them (using older version) or just wait for them to be obselete (2 years old in my case), this time I couldn't go back, as I've wrote above I've deleted them offline (corresponding directories) but some metadata seems to persist elsewhere and it still complains about them

**Comment by anasalkouz** [comment]
Yeah, I don't think deleting index directory will solve the problem, since you still have your index meta data stored somewhere else.
